cli/azd/pkg/pipeline/azdo_provider.go

```diff
@@ -702,17 +702,59 @@ func (p *AzdoCiProvider) credentialOptions(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	authType PipelineAuthType,
-) *CredentialOptions {
-	if authType == "" || authType == AuthTypeClientCredentials {
+	credentials *azcli.AzureCredentials,
+) (*CredentialOptions, error) {
+	// Default auth type to client-credentials for terraform
+	if infraOptions.Provider == provisioning.Terraform && authType == "" {
+		authType = AuthTypeClientCredentials
+	}
+
+	if authType == AuthTypeClientCredentials {
 		return &CredentialOptions{
 			EnableClientCredentials: true,
+		}, nil
+	}
+
+	// If not specified default to federated credentials
+	if authType == "" || authType == AuthTypeFederated {
```

Comment:
I think the only valid values for `PipelineAuthType` are `AuthTypeFederated` and `AuthTypeClientCredentials`.

Since L712 tests for `AuthTypeClientCredentials`, won't this test always evaluate to `true`?

Should we remove the init (and if we are worried about a future where another `PipelineAuthType` is added and we don't think to update this) add some sort of `panic` here, because it means we're in a state that we as the `azd` developers didn't anticipate?

Or are there still cases where we expect this test to fail and want to return the zero value as we do on L754?

Reply:
I copied/pasted the conditions/flow from the github provider XD (yes, I'm lazy).  Since gh-provider already works and resolves as expected, I didn't take much time in looking into this.

Let me keep it like this and I promise to enhance this condition for both (azdo and gh) while updating oidc support for terraform (as this conditions will be directly impacted)